apps/webapp/src/state/swap.ts

```diff
@@ -50,56 +55,73 @@ export const createSwapSlice = (): SliceCreator<SwapSlice> => (set, get) => {
+    initiateSwapTx: async () => {
       set(state => {
         state.swap.txInProgress = true;
       });
 
-      const { dismiss, update } = toastFn(buildingTxToast());
+      const swapTxToastId = toast(
+        ...buildingTxToast(undefined, undefined, 'Building swap transaction'),
+      );
 
       try {
         // build swap
         const swapPlan = await plan(await assembleSwapRequest(get().swap));
-        const swapTx = await authWitnessBuild(update, { transactionPlan: swapPlan });
+        const swapTx = await authWitnessBuild({ transactionPlan: swapPlan }, status =>
+          toast(...buildingTxToast(status, swapTxToastId, 'Building swap transaction')),
+        );
+        const swapTxHash = await getTxHash(swapTx);
+
+        await broadcast({ awaitDetection: true, transaction: swapTx }, status =>
+          toast(...broadcastingTxToast(swapTxHash, status, swapTxToastId)),
+        );
+
+        toast.success(
+          ...successTxToast(swapTxHash, undefined, swapTxToastId, 'Swap transaction succeeded 🎉'),
+        );
+        const swapClaimTxToastId = toast(
+          ...buildingTxToast(undefined, undefined, 'Building swap claim transaction'),
+        );
 
-        // begin broadcast
-        const swapBroadcast = broadcast(update, {
-          awaitDetection: true,
-          transaction: swapTx,
-        });
+        // start to build swap claim
+        const swapCommitment = getSwapCommitment(swapTx);
+        const swapClaimPlan = await plan(assembleSwapClaimRequest(swapCommitment));
 
         // a swap claim must be broadcast after any successful swap, to claim
         // the output of the swap.  Swap claims don't need authenticationData,
         // so use `witnessAndBuild`
         // @see https://protocol.penumbra.zone/main/zswap/swap.html#claiming-swap-outputs
-
-        // start to build swap claim
-        const swapCommitment = getSwapCommitment(swapTx);
-
-        const swapClaimPlan = await plan(assembleSwapClaimRequest(swapCommitment));
-        const swapClaimTx = await witnessBuild(update, { transactionPlan: swapClaimPlan });
+        const swapClaimTx = await witnessBuild({ transactionPlan: swapClaimPlan }, status =>
+          toast(...buildingTxToast(status, swapClaimTxToastId, 'Building swap claim transaction')),
+        );
+        const swapClaimTxHash = await getTxHash(swapClaimTx);
 
         // after swap broadcast success, emit the claim
-        await swapBroadcast;
-        const { txHash: swapClaimHash, detectionHeight: swapClaimDetectionHeight } =
-          await broadcast(update, {
-            transaction: swapClaimTx,
-          });
-
-        update(successTxToast(swapClaimHash, swapClaimDetectionHeight));
+        const { detectionHeight: swapClaimDetectionHeight } = await broadcast(
+          { awaitDetection: true, transaction: swapClaimTx },
+          status => toast(...broadcastingTxToast(swapClaimTxHash, status, swapClaimTxToastId)),
+        );
```

Comment:
this has broken the parallelism of the claim build during swap broadcast. now they happen sequentially

Reply:
We can't build a swap claim without the swap record in the database fyi:
Removed this here: https://github.com/penumbra-zone/web/pull/498#discussion_r1489442684